pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -260,10 +257,8 @@ public void run() {
           long segEndTime = realtimeSegment.getMaxTime();
 
           TimeUnit timeUnit = schema.getTimeFieldSpec().getOutgoingGranularitySpec().getTimeType();
-          Configuration configuration = new PropertyListConfiguration();
-          configuration.setProperty(IndexLoadingConfigMetadata.KEY_OF_LOADING_INVERTED_INDEX, invertedIndexColumns);
-          IndexLoadingConfigMetadata configMetadata = new IndexLoadingConfigMetadata(configuration);
-          IndexSegment segment = Loaders.IndexSegment.load(new File(resourceDir, segmentMetatdaZk.getSegmentName()), mode, configMetadata);
+          IndexSegment segment = Loaders.IndexSegment.load(new File(resourceDir, segmentMetatdaZk.getSegmentName()), mode,
+              realtimeTableDataManager.getIndexLoadingConfigMetadata());
```

Comment:
This has to be tested thoroughly. Not sure if there something else in the indexloading config that we wanted to avoid. I don't think there is, but I am not sure.